RallyHereIntegration/Source/RallyHereEditor/Private/RallyHereEditorLootWidget.cpp

```diff
@@ -0,0 +1,1123 @@
+					.AutoHeight()
+					[
+						SNew(SHorizontalBox)
+						+SHorizontalBox::Slot()
+						.Padding(FMargin(0.0f, 2.0f, 0.0f, 2.0f))
+						.AutoWidth()
+						[
+							SNew(STextBlock).Text(LOCTEXT("LootPropPreSalePricePointId", "Pre Sale Price Point Id: "))
+						]
+						+SHorizontalBox::Slot()
+						.AutoWidth()
+						[
+							SNew(SEditableTextBox)
+							.Text_Lambda([this]
+							{
+								return FText::FromString(LootUpdateRequest.GetPreSalePricePointId(FGuid()).ToString(EGuidFormats::DigitsWithHyphensLower));
+							})
+							.OnTextCommitted_Lambda([this](const FText& InNewText, ETextCommit::Type InCommitType)
+							{
+								FGuid InValue;
+								FGuid::Parse(InNewText.ToString(), InValue);
+								LootUpdateRequest.SetPreSalePricePointId(InValue);
+								IsDirty = true;
+								LootDirtyBox.Get()->Invalidate(EInvalidateWidget::Visibility);
+							})
+						]
+					]
+					+SVerticalBox::Slot()
+					.Padding(FMargin(0.0f, 2.0f, 0.0f, 2.0f))
+					.AutoHeight()
+					[
+						CustomData.ToSharedRef()
+					]
+					+SVerticalBox::Slot()
+					.Padding(FMargin(0.0f, 2.0f, 0.0f, 2.0f))
+					.AutoHeight()
+					[
+						SNew(SHorizontalBox)
+						+SHorizontalBox::Slot()
+						.AutoWidth()
+						[
+							SNew(SButton)
+							.Visibility_Lambda([this]
+							{
+								return IsNewLoot ? EVisibility::Visible : EVisibility::Collapsed;
+								})
+							.Text(LOCTEXT("CancelLootCreation", "Cancel New Loot"))
+							.OnClicked_Lambda([this]()
+								{
+									OnLootDeleteDelegate.ExecuteIfBound(Loot.Get()->GetLootId(), Loot);
+									return FReply::Handled();
+								})
+						]
+						+SHorizontalBox::Slot()
+						.AutoWidth()
+						[
+							SNew(SButton)
+							.Visibility_Lambda([this]
+								{
+									return !IsNewLoot ? EVisibility::Visible : EVisibility::Collapsed;
+								})
+							.Text(LOCTEXT("DeleteLootItem", "Delete Loot"))
+							.OnClicked(this, &SRallyHereEditorLootRow::OnDeleteLoot)
+						]
+					]
+				];
+
+	STableRow<TSharedPtr<FRHAPI_DevLoot>>::Construct(
+		STableRow<TSharedPtr<FRHAPI_DevLoot>>::FArguments()
+		.Content()
+		[
+			SNew(SHorizontalBox)
+			+SHorizontalBox::Slot()
+			.FillWidth(1.0f)
+			.Padding(FMargin(20.0f, 2.0f, 0.0f, 2.0f))
+			[
+				LootExpandableArea.ToSharedRef()
+			]
+		]
+		, OwnerTable);
+}
+
+#define COPY_OPTIONAL_VALUE(PropertyName, Source, Target) \
+	if (const auto Value = Source##Get##PropertyName##OrNull()) \
+	{ \
+		Target##Set##PropertyName(*Value); \
+	} \
+	else \
+	{ \
+		Target##Clear##PropertyName(); \
+	}
+
+#define COPY_OPTIONAL_VALUE_WITH_DEFAULT(PropertyName, Source, Target, DefaultValue) \
+	if (const auto Value = Source##Get##PropertyName##OrNull()) \
+	{ \
+		Target##Set##PropertyName(*Value); \
+	} \
+	else \
+	{ \
+		Target##Set##PropertyName(DefaultValue); \
+	}
+
+void SRallyHereEditorLootRow::HydrateLootUpdateRequest()
+{
+	IsDirty = false;
+	COPY_OPTIONAL_VALUE(CustomData, Loot.Get()->, LootUpdateRequest.)
```

Comment:
Should probably be GetValue() rather than Get()->

Reply:
Also passing the dot in the third param feels odd.  Generally with macros you would pass the object as the param in most cases (then macro would inject the operator)